### PR TITLE
Feat validation input

### PR DESCRIPTION
**Pull request by Sebastien-Ahkrin**
close: https://github.com/zakodium-oss/react-science/issues/346

### CODE REVIEW

File: src/components/forms/Input.tsx [hunk 1]
```diff
@@ -128,41 +142,111 @@ export interface InputProps extends InputHTMLAttributes<HTMLInputElement> {
...
+      style={{
+        color: hasErrorMessage
+          ? 'red'
+          : hasValidMessage || hasValidBoolean
+          ? 'green'
+          : undefined,
+      }}
+    >
+      {textToDisplay}
+    </p>
```
Comment (stropitek) [inline src/components/forms/Input.tsx:250]: This component has bugs and can be simplified (easier to explain voice to voice)

**Comment by stropitek** [comment]
Could you group the 3 stories which use subtext (error, valid and help) into one?

**Comment by stropitek** [comment]
image

`help message` -> `valid message`, it's a bit confusing otherwise

Inputs with labels + validation messages should look more like that:

image

Screenshot from https://ant.design/components/form#components-form-demo-complex-form-control

**Comment by stropitek** [comment]
Please ping me when you are finished

**Comment by Sebastien-Ahkrin** [comment]
> Please ping me when you are finished

i think, its done you can review :) @stropitek

**Comment by stropitek** [comment]
The alignement for small fields is incorrect:
image

I think the label could have a smaller font size

**Comment by Sebastien-Ahkrin** [comment]
> The alignement for small fields is incorrect: image
> 
> I think the label could have a smaller font size

fixed! :) Let me know if its good for you !

**Pull request merged**